README
Task-Devel-Essentials version 0.011
===================================

Task::Devel::Essentials is a bundle of modules useful for developers.

INSTALLATION

To install this bundle, just install it with an installer:

   cpan  Task::Devel::Essentials

The modules collected in this bundle are then installed as prerequisites.
Finally, your installer will do something equivalent to:

   perl Makefile.PL
   make
   make test
   make install

You can repeat the tests later to see if the modules of this bundle
are still present.

LICENSE AND COPYRIGHT

Copyright (c) 2013-2018 Martin Becker

This bundle is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.



Changes
Revision history for Task-Devel-Essentials

0.012   
        migrated from Module::Install to ExtUtils::MakeMaker

0.011   Thu, 20 Sep 2018 08:37:45 +0200
        upgraded more versions
        fixed unsafe @INC usage

0.009   Tue, 26 Jun 2018 14:36:33 +0200
        upgraded more versions

0.008   Tue, 26 Jun 2018 13:17:19 +0200
        upgraded versions

0.007   Fri, 20 Apr 2018 13:50:27 +0200
        upgraded versions
        sorted modules topologically
        added tidbits

0.006   Mon, 30 Oct 2017 12:20:51 +0100
        upgraded versions
        added more modules
        fixed version of CPAN

0.005   Fri, 20 Oct 2017 00:00:36 +0200
        fixed versions of modules newer in core than on CPAN

0.004   Wed, 18 Oct 2017 17:42:37 +0200
        upgraded versions
        added more modules
        dropped DBI

0.003   Fri, 30 May 2014 14:31:52 +0200
        added more modules
        added version numbers
        added tests

0.002   Mon, 03 Jun 2013 17:33:28 +0200
        added Module::Installer
        fixed license

0.001   Mon, 03 Jun 2013 15:31:46 +0200
        Initial version.



lib/Task/Devel/Essentials.pm
package Task::Devel::Essentials;

use 5.006;
use strict;
use warnings;

=head1 NAME

Task::Devel::Essentials - Bundle of modules useful for developers

=head1 VERSION

Version 0.011

=cut

our $VERSION = '0.011';

=head1 SYNOPSIS

  # to install a couple of modules useful for developers in one go,
  # execute one of the commands below:

  cpan Task::Devel::Essentials
  cpanm Task::Devel::Essentials
  cpanp i Task::Devel::Essentials

=head1 MODULES

=over 4

=item L<App::CPANTS::Lint>

=item L<Archive::Tar>

=item L<Archive::Zip>

=item L<Assert::Conditional>

=item L<Contextual::Return>

=item L<CPAN>

=item L<CPAN::Reporter>

=item L<Data::Dumper>

=item L<Data::Dumper::Simple>

=item L<Data::Printer>

=item L<DateTime>

=item L<Devel::Cover>

=item L<Devel::NYTProf>

=item L<Devel::StackTrace>

=item L<Digest::SHA>

=item L<Dist::Zilla>

=item L<Encode>

=item L<ExtUtils::CBuilder>

=item L<ExtUtils::MakeMaker>

=item L<File::HomeDir>

=item L<File::ShareDir>

=item L<File::Spec>

=item L<File::Temp>

=item L<File::Which>

=item L<Getopt::Long>

=item L<Inline>

=item L<IO::Compress::Base>

=item L<IO::Zlib>

=item L<Iterator::Diamond>

=item L<JSON>

=item L<Kwalify>

=item L<List::MoreUtils>

=item L<List::MoreUtils::XS> (optional)

=item L<LWP>

=item L<MIME::Base64>

=item L<MIME::Lite>

=item L<Modern::Perl>

=item L<Module::CoreList>

=item L<Module::CPANTS::Analyse>

=item L<Module::Install>

=item L<Module::Signature>

=item L<Module::Starter>

=item L<Module::Version>

=item L<Moo>

=item L<Moose>

=item L<Mouse>

=item L<Net::FTP>

=item L<Params::Check>

=item L<Parse::CPAN::Meta>

=item L<Perl::Critic>

=item L<Readonly>

=item L<Regexp::Common>

=item L<Role::Basic>

=item L<Scalar::Util>

=item L<Term::ReadKey>

=item L<Term::ReadLine>

=item L<Term::ReadLine::Gnu> (optional)

=item L<Test::Class>

=item L<Test::Differences>

=item L<Test::Exception>

=item L<Test::Fatal>

=item L<Test::Harness>

=item L<Test::MockModule>

=item L<Test::More>

=item L<Test::NoWarnings>

=item L<Test::Output>

=item L<Test::Pod>

=item L<Test::Pod::Coverage>

=item L<Test::Portability::Files>

=item L<Test::Regexp>

=item L<Test::Reporter::Transport::Metabase>

=item L<Test::Requires>

=item L<Test::Signature>

=item L<Test::Simple>

=item L<Test::Warn>

=item L<Test2::Suite>

=item L<Text::Glob>

=item L<Try::Tiny>

=item L<Type::Tiny>

=item L<YAML>

=item L<YAML::Syck>

=back

=head1 AUTHOR

Martin Becker, C<< <mhasch at cpan.org> >>

=head1 LICENSE AND COPYRIGHT

Copyright 2013-2018 Martin Becker.

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.

=cut

1;

__END__


Makefile.PL
# Copyright (c) 2013-2018 Martin Becker.  All rights reserved.
# This script is free software; you can redistribute it and/or modify it
# under the same terms as Perl itself.

BEGIN {
    $ENV{'NYTPROF'} = 'start=no';
}

use ExtUtils::MakeMaker 6.64;

my $module_filename = 'lib/Task/Devel/Essentials.pm';

WriteMakefile(
    NAME             => 'Task-Devel-Essentials',
    VERSION_FROM     => $module_filename,
    ABSTRACT_FROM    => $module_filename,
    LICENSE          => 'perl_5',
    AUTHOR           => 'Martin Becker <mhasch at cpan.org>'
    MIN_PERL_VERSION => '5.006',
    TEST_REQUIRES    => {
        'Test::More'                           => '0.82',
    },
    PREREQ_PM        => {
        'App::CPANTS::Lint'                    => '0.05',
        'Archive::Tar'                         => '2.32',
        'Archive::Zip'                         => '1.64',
        'Assert::Conditional'                  => '0.009',
        'CPAN'                                 => '2.16',
        'CPAN::Reporter'                       => '1.2018',
        'Contextual::Return'                   => '0.004014',
        'Data::Dumper'                         => '2.172',
        'Data::Dumper::Simple'                 => '0.11',
        'Data::Printer'                        => '0.40',
        'DateTime'                             => '1.50',
        'Devel::Cover'                         => '1.31',
        'Devel::NYTProf'                       => '6.06',
        'Devel::StackTrace'                    => '2.03',
        'Digest::SHA'                          => '6.02',
        'Dist::Zilla'                          => '6.012',
        'Encode'                               => '2.98',
        'ExtUtils::CBuilder'                   => '0.280230',
        'ExtUtils::MakeMaker'                  => '7.34',
        'File::HomeDir'                        => '1.004',
        'File::ShareDir'                       => '1.116',
        'File::Spec'                           => '3.75',
        'File::Temp'                           => '0.2308',
        'File::Which'                          => '1.22',
        'Getopt::Long'                         => '2.5',
        'IO::Compress::Base'                   => '2.081',
        'IO::Zlib'                             => '1.10',
        'Inline'                               => '0.80',
        'Iterator::Diamond'                    => '1.01',
        'JSON'                                 => '2.97001',
        'Kwalify'                              => '1.22',
        'LWP'                                  => '6.35',
        'List::MoreUtils'                      => '0.428',
        'MIME::Base64'                         => '3.15',
        'MIME::Lite'                           => '3.030',
        'Modern::Perl'                         => '1.20180901',
        'Module::CPANTS::Analyse'              => '0.96',
        'Module::CoreList'                     => '5.20180820',
        'Module::Install'                      => '1.19',
        'Module::Signature'                    => '0.83',
        'Module::Starter'                      => '1.75',
        'Module::Version'                      => '0.12',
        'Moo'                                  => '2.003004',
        'Moose'                                => '2.2011',
        'Mouse'                                => '2.005006',
        'Net::FTP'                             => '3.11',
        'Params::Check'                        => '0.38',
        'Parse::CPAN::Meta'                    => '2.150010',
        'Perl::Critic'                         => '1.132',
        'Readonly'                             => '2.05',
        'Regexp::Common'                       => '2017060201',
        'Role::Basic'                          => '0.13',
        'Scalar::Util'                         => '1.50',
        'Term::ReadKey'                        => '2.37',
        'Term::ReadLine'                       => '1.14',
        'Test2::Suite'                         => '0.000115',
        'Test::Class'                          => '0.50',
        'Test::Differences'                    => '0.64',
        'Test::Exception'                      => '0.43',
        'Test::Fatal'                          => '0.014',
        'Test::Harness'                        => '3.42',
        'Test::MockModule'                     => '0.170.0',
        'Test::More'                           => '1.302140',
        'Test::NoWarnings'                     => '1.04',
        'Test::Output'                         => '1.031',
        'Test::Pod'                            => '1.52',
        'Test::Pod::Coverage'                  => '1.10',
        'Test::Portability::Files'             => '0.09',
        'Test::Regexp'                         => '2017040101',
        'Test::Reporter::Transport::Metabase'  => '1.999010',
        'Test::Requires'                       => '0.10',
        'Test::Signature'                      => '1.11',
        'Test::Simple'                         => '1.302140',
        'Test::Warn'                           => '0.36',
        'Text::Glob'                           => '0.11',
        'Try::Tiny'                            => '0.30',
        'Type::Tiny'                           => '1.004002',
        'YAML'                                 => '1.26',
        'YAML::Syck'                           => '1.30',
    },
    META_MERGE => {
        prereqs => {
            runtime => {
                recommends => {
		    'List::MoreUtils::XS'                  => '0.428',
		    'Term::ReadLine::Gnu'                  => '1.35',
                },
            },
        },
        no_index => {
            directory => [
                'maint'
            ],
        }
    },
);
